subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/GroovyInteroperability.kt

```diff
@@ -121,6 +121,30 @@ class KotlinClosure2<in T : Any?, in U : Any?, V : Any>(
 }
 
 
+/**
+ * Adapts a ternary Kotlin function to a ternary Groovy [Closure].
+ *
+ * @param T the type of the first argument.
+ * @param U the type of the second argument.
+ * @param V the type of the third argument.
+ * @param R the return type.
+ * @param function the function to be adapted.
+ * @param owner optional owner of the Closure.
+ * @param thisObject optional _this Object_ of the Closure.
+ *
+ * @see [Closure]
+ */
+class KotlinClosure3<in T : Any?, in U : Any?, in V : Any?, R : Any>(
+    val function: (T, U, V) -> R?,
+    owner: Any? = null,
+    thisObject: Any? = null
+) : Closure<R?>(owner, thisObject) {
```

Comment:
Do you want to expose constructors for these directly instead of the `of` methods like you were doing elsewhere?

Reply:
Yes